src/main/java/com/capstone/controller/RoadmapController.java

```diff
@@ -39,4 +40,11 @@ public ResponseEntity<String> updateLearnerProgress(@RequestBody AtomProgressReq
         return ResponseEntity.status(HttpStatus.OK).body("response");
     }
 
+    @PostMapping("/recalculate-progress")
+    @Operation(summary = "Recalculate", description = "This end point recalculates learner's progress for consistency")
+    public ResponseEntity<String> recalculateLearnerProgress(@RequestBody RecalculateProgressRequestDto dto) {
+        learnerProgressService.recalculateAndUpdateLearnerRoadmap(dto);
+        return ResponseEntity.status(HttpStatus.OK).body("response");
```

Comment:
If the response body is never validated on the FE side, then simply returning a success response will suffice.

```Java
  return ResponseEntity.status(HttpStatus.OK).build();
  // OR
  return ResponseEntity.ok().build();
```